ecommerce/api.py

```diff
@@ -162,3 +166,40 @@ def make_reference_id(order):
             A reference number for use with CyberSource to keep track of orders
     """
     return "MM-{}-{}".format(settings.CYBERSOURCE_REFERENCE_PREFIX, order.id)
+
+
+def get_new_order_by_reference_number(reference_number):
```

Comment:
i meant that the `Order` object is assumed to exist when this method is called. otherwise line 203 would fail every time. after reading your comment, though, this seems like a good name